constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -33,166 +45,382 @@ package k8scel
+		d.filterCompiler.Compile(convertV1Alpha1MessageExpressions(validatorCode.Spec.Validations), celVars, celAPI.PerCallLimit),
+		failurePolicy,
+		nil,
+	)
+
+	d.mux.Lock()
+	defer d.mux.Unlock()
+	d.validators[ct.GetName()] = validator
+	return nil
+}
+
+func (d *Driver) RemoveTemplate(ctx context.Context, ct *templates.ConstraintTemplate) error {
+	d.mux.Lock()
+	defer d.mux.Unlock()
+	delete(d.validators, ct.GetName())
+	return nil
+}
+
+func (d *Driver) AddConstraint(ctx context.Context, constraint *unstructured.Unstructured) error {
+	return nil
+}
+
+func (d *Driver) RemoveConstraint(ctx context.Context, constraint *unstructured.Unstructured) error {
+	return nil
+}
+
+func (d *Driver) AddData(ctx context.Context, target string, path storage.Path, data interface{}) error {
+	return nil
+}
+
+func (d *Driver) RemoveData(ctx context.Context, target string, path storage.Path) error {
+	return nil
+}
+
+func (d *Driver) Query(ctx context.Context, target string, constraints []*unstructured.Unstructured, review interface{}, opts ...drivers.QueryOpt) (*drivers.QueryResponse, error) {
+	cfg := &drivers.QueryCfg{}
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	d.mux.RLock()
+	defer d.mux.RUnlock()
+
+	var statsEntries []*instrumentation.StatsEntry
+
+	arGetter, ok := review.(ARGetter)
+	if !ok {
+		return nil, errors.New("cannot convert review to ARGetter")
+	}
+	aRequest := arGetter.GetAdmissionRequest()
+	request, err := NewWrapper(aRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	results := []*types.Result{}
+
+	for _, constraint := range constraints {
+		evalStartTime := time.Now()
+		// FRICTION/design question: should parameters be created as a "mock" object so that users don't have to type `params.spec.parameters`? How do we prevent visibility into other,
+		// non-parameter fields, such as `spec.match`? Does it matter? Note that creating a special "parameters" object means that we'd need to copy the constraint contents to
+		// a special "parameters" object for on-server enforcement with a clean value for "params", which is non-ideal. Could we provide the field of the parameters object and limit scoping to that?
+		// Then how would we implement custom matchers? Maybe adding variable assignments to the Policy Definition is a better idea? That would at least allow for a convenience handle, even if
+		// it doesn't scope visibility.
+
+		// template name is the lowercase of its kind
+		validator := d.validators[strings.ToLower(constraint.GetKind())]
+		if validator == nil {
+			return nil, fmt.Errorf("unknown constraint template validator: %s", constraint.GetKind())
+		}
+		versionedAttr := &admission.VersionedAttributes{
+			Attributes:         request,
+			VersionedKind:      request.GetKind(),
+			VersionedOldObject: request.GetOldObject(),
+			VersionedObject:    request.GetObject(),
+		}
+		response := validator.Validate(ctx, versionedAttr, constraint, 10000000)
```

Comment:
what is `10000000`?

Reply:
https://github.com/open-policy-agent/frameworks/pull/331#discussion_r1205926676
looks like we missed this to be a const